sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -253,9 +253,8 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
    */
   async renewSessionLock(options?: OperationOptions): Promise<Date> {
     this._throwIfReceiverOrConnectionClosed();
-
+    this._throwIfMessageSessionDoesntExist();
```

Comment:
Do we really need to check this at all? Since these operations are on the management link, they technically don't need the `messageSession`. We had this in Track 1 because `createSessionReceiver()` does not guarantee getting a lock on the session. Since that has now changed, can we skip this check altogether for all of the management link operations? I would expect the service to throw an error if the lock was lost and the operation cannot be performed.

This would apply to peek, set/get state, receiveDeferredMessages as well

Reply:
Fine.

Reply:
One thing to check would be as to how these methods behave if the lock is lost.
If they get a meaningful error back, then we are good

Reply:
Tried the receiveDeferredMessages API
  - After the locked time, `receiveDeferredMessages` threw an error from `_throwIfReceiverOrConnectionClosed();`
    `Error: The receiver for session "my-session" in "unpartitioned-queue-sessions" has been closed and can no longer be used. Please create a new receiver using the "createSessionReceiver" method on the ServiceBusClient.`
  - Reason being receiver is closed(`receiver.isClosed = true`), 
    `isClosed` getter has this boolean check  
     `this._isClosed || (this.sessionId ? !this._context.messageSessions[this.sessionId] : false)`
  - `this._context.messageSessions[this.sessionId]` is undefined in the above boolean check and hence the boolean returned is true, 
     `MessageSession._deleteFromCache` removes it from `_context.messageSessions`
     `_onAmqpClose` (for the "receiver_close" event) listener calls `MessageSession.close` and that calls `deleteFromCache`.

The workflow looks good and throwing error seems fine to me!!
Updated the error messages though.

Reply:
The operations on the $management link like peek, set/get state, receive deferred messages don't really need the message session. What happens when we remove the check on the message session and try these operations after the lock is lost? What is the precedence in .Net?

Reply:
Well, disabled `_throwIfReceiverOrConnectionClosed`, and the `receiveDeferredMessages` ran into the following error after the lock is expired.

> MessagingError: The service was unable to process the request; please retry the operation. For more information on exception types and proper exception handling, please refer to http://go.microsoft.com/fwlink/?LinkId=761101 TrackingId:xxx-xxx-xxxx-xxx, SystemTracker:\<namespace-name\>:Queue:unpartitioned-queue-sessions, Timestamp:2020-07-06T20:19:27

Reply:
> What is the precedence in .Net?

SessionLockLost error is thrown for all the session related operations, not just renewSessionLock.
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs#L939

Reply:
Alright then. Let's go ahead with us throwing the SessionLockLostError.

Reply:
One more observation...

Tried the same experiment with `getState()` instead of `receiveDeferredMessages()` (by disabling `_throwIfReceiverOrConnectionClosed`),
- Error thrown
  `MessagingError: The session lock has expired on the MessageSession. Accept a new MessageSession. TrackingId:xxxxxxxx-xxxx-xxxx-xxxxxxxx, SystemTracker:NoSystemTracker, Timestamp:2020-07-07T21:02:42`

Following .NET's precedence by throwing SessionLockLost error still seems fine. This also makes the error thrown for the session related operations to be consistent.